### DISCUSSION TITLE
Feature Request: Scripting/Automation

### DISCUSSION THREAD

**Discussion by Wahooney** [4sval/FModel#495] [Q&A] [answered]
I've had a few cases where I've needed to export a lot of assets, but I don't have a direct link to the asset in question, ie. I have a list of package names of fortnite weapons and I need to export the large icons. This is a tedious task and some sort of scripting/automation to speed it up would be amazing, eg. Take this list of packages, find the "LargeIcon" in json, export it and append the filename to this file.

Scripting (hopefully python*) would be ideal as it would give the most flexibility, but at this point I'd take anything.

The python bindings could be very basic to allow for very powerful automation. Example code:

```
import fmodel

my_packages = [list of packages]

for p in my_packages:
  package = fmodel.resolve_package_json(p)
  icon_package = fmodel.resolve_package_json(package['LargeIcon']['AssetPathName'])
  save_path = fmodel.save_texture_png(icon_package)

  # do the python-stuff to write to file which won't require any action from FModel so I won't elaborate :P
```

**Comment by 4sval** [comment, marked as answer]
https://github.com/FabianFG/CUE4Parse

**Comment by Wahooney** [comment]
Thank you!

Thanks to your recommendation, Avalonia, IronPython and ~10 hours work, I now have exactly what I was looking for 🥳 

In my tests I can export ~90 Item Icons in different plugins / folders / game features in ~4 seconds, where hunting them down manually took ~4 hours.

Someday I hope to make it a little friendlier and release it for general consumption.
